### PR TITLE
[PUB-3532]: Update to Jest 27

### PR DESCRIPTION
**Pull request by hamstu**
### Description

https://buffer.atlassian.net/browse/PUB-3532

### Best Practices Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] I have tested different user stories.
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

### ⚠️ Pre-deploy Checklist

- [ ] I have manually loaded the URL of the staging deployment and confirmed it's working (see below**)
- [ ] I have confirmed that the Cypress E2E test is passing (_Note: we have a known issue where it doesn't run on some PRs. Hopefully this will be fixed soon._)
- [ ] I have confirmed that all other tests are passing.

** To visit the URL of the staging deployment, please vlick the link in the bot comment, or look under checks at the bottom of this PR and click "Details" next to `bufferbotbrains/cicd-buffer-publish`
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -188,16 +188,16 @@
...
     "eslint-plugin-jsx-a11y": "6.2.3",
     "eslint-plugin-prettier": "^3.1.4",
     "eslint-plugin-react": "7.19.0",
     "eslint-plugin-react-hooks": "2.5.0",
     "eslint-plugin-testing-library": "^3.10.0",
     "inquirer": "7.1.0",
-    "jest": "24.9.0",
-    "jest-axe": "^3.5.0",
-    "jest-styled-components": "^7.0.3",
+    "jest": "27.0.4",
```
Comment (MayaUribe) [inline package.json:198]: 🤩

**Review by MayaUribe — APPROVED** [review]
Thank you for upgrading Jest @hamstu!! Nice one! 🌮

**Comment by hamstu** [comment]
@MayaUribe Yep, good one! I figured it would fail, but hoping it's not too hard to fix. I'll take a look 😄

**Comment by anafilipadealmeida** [comment]
Closing this as Mw already updated it 🙂

**Pull request closed without merging**